compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -274,33 +272,25 @@ struct GPUSetEncodingOpLoweringConversion
       rewriter.replaceOp(encodingOp, packOp->getResult());
       return success();
     }
-    SmallVector<int64_t> innerTiles = maybeEncodingInfo->innerTileSizes;
 
-    // TODO(hanchung): Add a util to the encoding attribute, so we don't need
-    // the map_to_vector method here.
-    auto loc = encodingOp.getLoc();
+    Location loc = encodingOp.getLoc();
 
     // Create expand_shape op to tile the innermost two dimensions.
     int origRank = encodingOp.getSourceType().getRank();
-    SmallVector<int64_t> expandShapeShape(packOp->getDestType().getShape());
-    expandShapeShape.truncate(origRank);
+    SmallVector<int64_t> expandShapeShape(
+        packOp->getDestType().getShape().take_front(origRank));
     expandShapeShape.append(
         getExpandedTileShape(maybeEncodingInfo->swizzle->expandShape));
-
-    auto expandShapeType = RankedTensorType::get(
-        expandShapeShape, encodingOp.getSourceType().getElementType());
+    RankedTensorType expandShapeType =
+        encodingOp.getSourceType().clone(expandShapeShape);
 
     SmallVector<ReassociationIndices> reassociation = getReassociationIndices(
         origRank, maybeEncodingInfo->swizzle->expandShape);
     auto expandShapeOp = rewriter.create<tensor::ExpandShapeOp>(
         loc, expandShapeType, packOp->getResult(), reassociation);
 
-    // create linalg.transpose on expandShapeShape
-
-    SmallVector<int64_t> transposePerm;
-    for (int i = 0; i < origRank; ++i) {
-      transposePerm.push_back(i);
-    }
+    SmallVector<int64_t> transposePerm =
+        llvm::to_vector(llvm::seq<int64_t>(0, origRank));
```

Comment:
You can use std::iota as well

Reply:
I considered it before! The issue is that you need to allocate (or resize) the vector before using the iota. Thus, I end up with using to_vector + seq.

E.g.,

```cpp
SmallVector<int64-t> transposePerm(origRank);
std::iota(transposePerm.begin(), transposePerm.end(), 0);
```